File: de.wwu.md2.android.lib/src/main/java/de/uni_muenster/wi/md2library/controller/eventhandler/interfaces/Md2ContentProviderEventHandler.java
package de.uni_muenster.wi.md2library.controller.eventhandler.interfaces;

/**
 * Content provider Event handler interface that is implemented by all
 * content provider event handlers.
 * <p/>
 * Created on 11/08/2015
 *
 * @author Fabian Wrede
 * @version 1.0
 * @since 1.0
 */
public interface Md2ContentProviderEventHandler extends Md2EventHandler {
    /**
     * On change.
     *
     * @param attribute the changed attribute
     */
    void onChange(String attribute);
}

File: de.wwu.md2.framework2/res/resources/ios/lib/controller/eventhandler/MD2ContentProviderEventHandler.swift
//
//  MD2ContentProviderEventHandler.swift
//  md2-ios-library
//
//  Created by Christoph Rieger on 23.07.15.
//  Copyright (c) 2015 Christoph Rieger. All rights reserved.
//

/// Interface for content provider event handlers.
protocol MD2ContentProviderEventHandler: MD2EventHandler {
    
    /**
        Register an action to a content provider event handler.
    
        :param: action The action to execute in case of an event.
        :param: contentProvider The content provider that the action should be bound to.
        :param: attribute The content provider attribute the action should be bound to.
    */
    func registerAction(action: MD2Action, contentProvider: MD2ContentProvider, attribute: String)
    
    /**
        Unregister an action from a content provider event handler.
    
        :param: action The action to remove.
        :param: contentProvider The content provider the action was registered to.
        :param: attribute The content provider attribute the action was registered to.
    */
    func unregisterAction(action: MD2Action, contentProvider: MD2ContentProvider, attribute: String)
    
}